src/modules/kubernetes.rs

```diff
@@ -0,0 +1,154 @@
+
+    if current_ctx.is_empty() {
+        return None;
+    }
+
+    let mut ns = "";
+
+    if let Some(contexts) = conf["contexts"].as_vec() {
+        for ctx in contexts {
+            if ctx["name"].as_str().unwrap_or("") == current_ctx {
+                ns = ctx["context"]["namespace"].as_str().unwrap_or("");
+                break;
+            }
+        }
+    }
+
+    Some((current_ctx.to_string(), ns.to_string()))
+}
+
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let filename = match env::var("KUBECONFIG") {
+        Ok(path) => path::PathBuf::from(path),
+        Err(_) => dirs::home_dir()?.join(".kube").join("config"),
+    };
+
+    let contents = utils::read_file(filename).ok()?;
+
+    match get_kube_context(&contents) {
+        Some(kube_cfg) => {
+            const KUBE_CHAR: &str = "☸ ";
+
+            let (kube_ctx, kube_ns) = kube_cfg;
+
+            let mut module = context.new_module("kubernetes");
+
+            let module_style = module
+                .config_value_style("style")
+                .unwrap_or_else(|| Color::Cyan.bold());
+            module.set_style(module_style);
+            module.get_prefix().set_value("on ");
+
+            module.new_segment("symbol", KUBE_CHAR);
+            module.new_segment("context", &kube_ctx);
+            if kube_ns != "" {
+                module.new_segment("namespace", &format!(" ({})", kube_ns));
+            }
+            Some(module)
+        }
+        None => None,
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn parse_empty_config() {
+        let input = "";
+        let result = get_kube_context(&input);
+        let expected = None;
+
+        assert_eq!(result, expected);
+    }
+
+    #[test]
+    fn parse_no_config() {
+        let input = r#"
+apiVersion: v1
+clusters: []
+contexts: []
+current-context: ""
+kind: Config
+preferences: {}
+users: []
+"#;
+        let result = get_kube_context(&input);
+        let expected = None;
+
+        assert_eq!(result, expected);
+    }
+    #[test]
+    fn parse_only_context() {
+        let input = r#"
+apiVersion: v1
+clusters: []
+contexts:
+- context:
+    cluster: test_cluster
+    user: test_user
+  name: test_context
+current-context: test_context
+kind: Config
+preferences: {}
+users: []
+"#;
+        let result = get_kube_context(&input);
+        let expected = Some(("test_context".to_string(), "".to_string()));
+
+        assert_eq!(result, expected);
+    }
+
+    #[test]
+    fn parse_context_and_ns() {
+        let input = r#"
+apiVersion: v1
+clusters: []
+contexts:
+- context:
+    cluster: test_cluster
+    user: test_user
+    namespace: test_namespace
+  name: test_context
+current-context: test_context
+kind: Config
+preferences: {}
+users: []
+"#;
+        let result = get_kube_context(&input);
+        let expected = Some(("test_context".to_string(), "test_namespace".to_string()));
+
+        assert_eq!(result, expected);
+    }
```

Comment:
Nice test setup. I would suggest to add one more test with multiple contexts. Let me lend you one similar to mine 😉 
```
apiVersion: v1
clusters: []
contexts:
- context:
    cluster: arn:aws:eks:eu-west-1:123456789:cluster/eks-cluster-eu-west-1-dev
    user: arn:aws:eks:eu-west-1:123456789:cluster/eks-cluster-eu-west-1-dev
  name: DEV
- context:
    cluster: arn:aws:eks:eu-west-1:123456789:cluster/eks-cluster-eu-west-1-prod
    user: arn:aws:eks:eu-west-1:123456789:cluster/eks-cluster-eu-west-1-prod
  name: PROD
current-context: DEV

```
As I'm writing this, I notice that namespace detection would not work with my setup. I don't have one namespace per context but use multiple dev namespaces within the dev context. Not sure if it's best practice to include the namespace in the context? I have to say I like not having to switch so often and if I do it's to get to PROD. But yeah I see that in this case it's impossible to show the namespace, so nvm.

Would still add the test 🙃

Reply:
And maybe we could note in the documentation that namespace detection only works if namespaces are bound to contexts

Reply:
Have added a test for this and updated the docs.